datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -59,8 +59,8 @@ def compile(
         query = self.query_data.get('query')
         if not query:
             raise ValueError('field `query` for {} is required'.format(query_name))
-        elif not isinstance(query, str):
-            raise ValueError('field `query` for {} must be a string'.format(query_name))
+        elif not isinstance(query, (str, dict)):
+            raise ValueError('field `query` for {} must be a string or a mapping'.format(query_name))
```

Comment:
Let's allow anything for ourselves, but keep error for custom queries:

```suggestion
        elif query_name.startswith('custom query #') and not isinstance(query, str):
            raise ValueError('field `query` for {} must be a string'.format(query_name))
```

https://github.com/DataDog/integrations-core/blob/63ca09196d515c95deaa88bbab9fd6cf0452b801/datadog_checks_base/datadog_checks/base/utils/db/core.py#L82-L86